### ISSUE TITLE
ListVaultsRequest Returns Max of 10

### ISSUE THREAD

**Issue by wjdavis5** [aws/aws-sdk-net#895]
Making this call:
`var request = new ListVaultsRequest("-", vaultResponse.Marker, 1000);`
returns a maximum of 10 items. If I put in 1 - 9 it will return the requested number, however, anything above 10 only returns 10 items.

## Expected Behavior
To return the specified number of items, or 1000 (not 10)

From the documentation:

> By default, this operation returns up to 1,000 items. If there are more vaults to
> list, the response <code>marker</code> field contains the vault Amazon Resource Name
> (ARN) at which to continue the list with a new List Vaults request; otherwise, the
> <code>marker</code> field is <code>null</code>. To return a list of vaults that begins
> at a specific vault, set the <code>marker</code> request parameter to the vault ARN
> you obtained from a previous List Vaults request. You can also limit the number of
> vaults returned in the response by specifying the <code>limit</code> parameter in
> the request. 

## Current Behavior

Returns a maximum of 10 items.

I'm not sure what is sensitive here so I redacted anything I thought may be.
Logs:

```
2018-03-08 08:22:38,535 DEBUG: Double encoded /-/vaults with endpoint https://glacier.us-west-2.amazonaws.com/ for canonicalization: /-/vaults
2018 - 03 - 08 08 : 22 : 39, 097 DEBUG : Received response(truncated to 1024 bytes) : [{
		"Marker" : "REDACTED",
		"VaultList" : [{
				"CreationDate" : "2016-06-23T05:57:11.393Z",
				"LastInventoryDate" : "2017-04-04T17:28:40.806Z",
				"NumberOfArchives" : 7383,
				"SizeInBytes" : 2467122050,
				"VaultARN" : "REDACTED,
				"VaultName" : "REDACTED"
			}, {
				"CreationDate" : "2015-09-16T20:03:51.737Z",
				"LastInventoryDate" : "2016-09-10T08:17:13.050Z",
				"NumberOfArchives" : 6,
				"SizeInBytes" : 200143,
				"VaultARN" : "REDACTED",
				"VaultName" : "REDACTED"
			}, {
				"CreationDate" : "2015-09-16T20:06:33.261Z",
				"LastInventoryDate" : "2016-09-10T08:17:56.755Z",
				"NumberOfArchives" : 1168,
				"SizeInBytes" : 213385560,
				"VaultARN" : "REDACTED",
				"VaultName" : "REDACTED"
			}, {
				"CreationDate" : "2015-09-16T20:09:59.057Z",
				"LastInventoryDate" : "2016-09-10T08:16:55.746Z",
				"NumberOfArchives" : 585,
				"SizeInBytes" : 82825218,
				"VaultARN" : "REDACTED]
2018-03-08 08:22:39,103 INFO : Request metrics: AsyncCall = False; CanonicalRequest = GET\n/-/vaults\nlimit=100\nhost:glacier.us-west-2.amazonaws.com\nuser-agent:aws-sdk-dotnet-45/3.3.2.2 aws-sdk-dotnet-core/3.3.21.15 .NET_Runtime/4.0 .NET_Framework/4.0 OS/Microsoft_Windows_NT_6.2.9200.0 ClientSync\nx-amz-content-sha256:REDACTED\nx-amz-date:20180308T132238Z\nx-amz-glacier-version:2012-06-01\n\nhost;user-agent;x-amz-content-sha256;x-amz-date;x-amz-glacier-version\REDACTED; StringToSign = AWS4-HMAC-SHA256\n20180308T132238Z\n20180308/us-west-2/glacier/aws4_request\REDACTED; ServiceName = Amazon.Glacier; ServiceEndpoint = https://glacier.us-west-2.amazonaws.com/; MethodName = ListVaultsRequest; StatusCode = OK; BytesProcessed = 2616; AWSRequestID = 2wlh2iTmT11R8jTZfb6ZTbhQyg7W8QJfpb4wn8ptUnjRvVE; CredentialsRequestTime = 00:00:00.0013664; RequestSigningTime = 00:00:00.0210118; HttpRequestTime = 00:00:00.5115086; ResponseUnmarshallTime = 00:00:00.0096159; ResponseProcessingTime = 00:00:00.0138258; ClientExecuteTime = 00:00:00.5890660;
```

## Steps to Reproduce (for bugs)
```
var client = new AmazonGlacierClient(accessKey, privateKey, endpoint);
                var vaultResponse = new ListVaultsResponse();

                do
                {
                    var request = new ListVaultsRequest("-", vaultResponse.Marker, 100); 
                    vaultResponse = client.ListVaults(request);
                    foreach (var vault in vaultResponse.VaultList) //Only 10 items here
                    {
                        DoStuff(vault);
                    }
                } while (vaultResponse.Marker != null);

                _vaultsFinishedLoading = true;
```
## Context
Trying to list all the vaults in a region

## Your Environment

* AWSSDK.Core version used: v3.3.2.2
* Service assembly and version used: 
* Operating System and version: Windows 10
* Visual Studio version:  Visualt Studio 2017 ( 15.6.0)
* Targeted .NET platform:  4.6.2

## .NET Core Info
This is .NET 4.6.2

[logging]: http://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-other.html

**Comment by steveataws** [comment]
Repro'd issue. 

A review of the SDK code to see if there were any statements that were overriding the input request max, nothing found.

We think this may be a service issue with incorrect docs. We've opened an internal support ticket and will update this issue when we get a response.

**Comment by steveataws** [comment]
Heard back from the service team that the max limit per call is indeed 10 and not 1000.

**Comment by wjdavis5** [comment]
Hi @steveataws  - thanks for looking. There are tonnes of references to 1k all throughout the glacier documentation.  When I do this:

`aws glacier list-vaults --account-id - --max-items 20`

It does indeed return 20 items. Shouldnt the .net api be using the same amount?

**Comment by steveataws** [comment]
All sdks and command line tools generate from the same models so they'll all have the same, incorrect, documentation regarding the 1K figure. Now, why the cli is able to return more than 10 I can't answer (don't work on that tool).

I will however point out the inconsistency with the service team :-)

**Comment by wjdavis5** [comment]
Hello, any updates on this?

**Comment by wjdavis5** [comment]
Hello, any updates on this?

**Comment by steveataws** [comment]
Apologies, I got distracted with other issues.

I just checked the internal ticket I raised on your behalf and see no recent activity, so I think the service team are still trying to determine if a doc update will suffice or whether an actual service update is needed. I pinged the ticket to see if I can get an updated response.

From the command line tools perspective, I'm pretty sure they are simply making calls asking for up to 10 each time.
